File: Java/[2 sem] 2DWorldSimulator/code/Guarana.java
public class Guarana extends Plant {
    private static final int GUARANA_STRENGTH = 0;
    private static final int GUARANA_INITIATIVE = 0;
    private static final char GUARANA_SYMBOL = '^';

    public Guarana(int x, int y, World world) {
        super(x, y, world, GUARANA_STRENGTH, GUARANA_INITIATIVE, GUARANA_SYMBOL, 0);
    }

    @Override
    public Organism collision(Organism attacker) {
        attacker.setStrength(attacker.getStrength() + 3);
        return attacker;
    }

    @Override
    public String getSpecieName() {
        return "Guarana";
    }
}

File: C(++)/[2 sem] 2DWorldSimulator/code/src/worldObjects/plants/Guarana.cpp
//
// Created by szymo on 4/18/2023.
//

#include ".headers/Guarana.h"

#define GUARANA_STRENGTH 0
#define GUARANA_INITIATIVE 0
#define GUARANA_SYMBOL '^'

Guarana::Guarana(int x, int y, World *world) : Plant(x, y, world, GUARANA_STRENGTH, GUARANA_INITIATIVE, GUARANA_SYMBOL) {}

Guarana::~Guarana() = default;

Organism* Guarana::collision(Organism *attacker) {
    attacker->setStrength(attacker->getStrength() + 3);
    return attacker;
}

std::string Guarana::getSpecieName() const {
    return "Guarana";
}